target/linux/ath79/dts/qca9557_extreme_ap3805i.dts

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "extreme,ap3805i", "qca,qca9557";
+	model = "Extreme Networks AP3805i";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_amber;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_amber;
+		label-mac-device = &eth0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power_green: led_power_green {
+			label = "ap3805:green:power";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_amber: led_power_amber {
+			label = "ap3805:amber:power";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan5_green {
+			label = "ap3805:green:wlan5";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};	
+
+		led_wlan2_green {
+			label = "ap3805:green:wlan2";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		hw_margin_ms = <20000>;
+		always-running;
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&wmac {
+	status = "okay";
+
+	qca,no-eeprom;
+};
+
+&pcie0 {
+	status = "okay";
+
+	wifi@0,0,0 {
+		compatible = "qcom,ath10k";
+		reg = <0x0 0 0 0 0>;
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy5: ethernet-phy@5 {
+		reg = <5>;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy5>;
+	phy-mode = "rgmii-id";
+
+	pll-data = <0x82000000 0x80000101 0x80001313>;
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-enabled = <1>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot-bak";
+				reg = <0x0 0x80000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "u-boot";
+				reg = <0x080000 0x80000>;
+				read-only;
+			};
+
+			partition@100000 {
+				label = "cfg1";
+				reg = <0x100000 0x40000>;
+				read-only;
+			};
+
+			partition@140000 {
+				label = "cfg2";
+				reg = <0x140000 0x40000>;
+				read-only;
+			};
+
+			partition@180000 {
+				label = "nvram4";
+				reg = <0x180000 0x40000>;
+			};
+
+			partition@1c0000 {
+				label = "nvram3";
+				reg = <0x1c0000 0x40000>;
+			};
+
+			partition@200000 {
+				label = "nvram2";
+				reg = <0x200000 0x40000>;
+			};
+
+			partition@240000 {
+				label = "nvram1";
+				reg = <0x240000 0x40000>;
+			}; 
+
+			partition@2280000 {
```

Comment:
So, what's between?

Reply:
Factory images, Pri+Sec.
And between "Firmware" and "ART" is some Siemens CA certificates.

Reply:
Well, then please add them here.

Reply:
I was unsure on how/if i should do that. Is the best practice to att it to the "firmware"-partition or just let them be their own RO partitions so users more easily can go back to stock? As the original firmware can be hard to come by without proper Extreme-licences i decided on this approach.

Reply:
General OpenWrt policy is to keep everything in a state that you can return to stock firmware, i.e. no overwriting of vendor-partitions should take place (unless cases like a config partition that is recreated by the vendor firmware anyway, like for some TP-Link devices). But adding the partitions to the list as read-only is common practice, so the user can easily read from them if desired.